File: final-project/src/main/java/com/example/final_project/entity/Coupon.java
package com.example.final_project.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter @Setter
public class Coupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String code;
    private double discountPrice;
    private int quantity;
    private boolean active;
}

File: lib/models/Coupon.dart
class Coupon {
  final int id;
  final String code;
  final double discountPrice;
  final int quantity;
  final bool active;

  Coupon({
    required this.id,
    required this.code,
    required this.discountPrice,
    required this.quantity,
    required this.active,
  });

  factory Coupon.fromJson(Map<String, dynamic> json) {
    return Coupon(
      id: json['id'],
      code: json['code'],
      discountPrice: (json['discountPrice'] ?? 0).toDouble(),
      quantity: json['quantity'],
      active: json['active'],
    );
  }
}
